### DISCUSSION TITLE
20230308: Word-level timestamps visualized

### DISCUSSION THREAD

**Discussion by glangford** [openai/whisper#1072] [Q&A] [answered]
I have experimented with a few videos with the latest release, and wanted to share a short (20s) example of what word level timestamps look like. Note the flashing text that occurs on occasion; this seems to correspond to .srt lines that have no underscores (an artifact of word level emphasis in the transcript). This effect occurs with both hard and soft subtitles added to a video. 

https://user-images.githubusercontent.com/3035114/224117897-6e7b9263-d796-4c48-99d1-86c343cfae8c.mp4

**Comment by glangford** [comment]
..but the timing of words is great, and it nails the spelling of the speaker's name!

**Comment by jongwook** [comment, marked as answer]
This is definitely not intended, I will look for any bugs in the code that writes underlined text. I should also consider making the underscoring optional 😅

**Reply by ryanheise** [reply]
The problem is not in the SRT generation, actually the result data contains segments where the text and the words in the segment do not match each other. The SRT generator uses the "words" when a word is spoken, but falls back to "text" when there is a pause or gap between words. This is where you see a flash, because the two pieces of data don't match.

**Reply by ryanheise** [reply]
The flashing is fixed by #1087 .

**Comment by glangford** [comment]
Here is the .srt, hopefully with some good clues - note that lines 4, 12 and 16 are odd and are inserted right after a punctuation mark - a comma, a period with "Dr." and the period at the end of the sentence. 

[twp-intro.srt.txt](https://github.com/openai/whisper/files/10934884/twp-intro.srt.txt)

**Comment by dgoryeo** [comment]
@glangford , what options/hyper-parameters did you use for the subs of the demo video?

**Reply by glangford** [reply]
`--verbose False --output_format srt --fp16 False --model large --word_timestamps True`

**Comment by Honghe** [comment]
@glangford Which video player do you use that support word-level underline subtitle?

**Reply by glangford** [reply]
It works with Quicktime Player on Mac and VLC media player, I haven't tested other players
